### PR TITLE
workaround: Extend the condition and add on other places too

### PR DESCRIPTION
**Pull request by dzedro**
- Related ticket: https://bugzilla.suse.com/show_bug.cgi?id=1195351 #14159 #14172
- Verification run:
https://openqa.suse.de/tests/8178323#step/patch_and_reboot/10 PC
https://openqa.suse.de/tests/8178331#step/patch_and_reboot/10 PC
https://openqa.suse.de/tests/8178551 HA
https://openqa.suse.de/tests/8174175#step/update_install/126 HA
https://openqa.suse.de/tests/overview?distri=sle&version=12-SP5&build=%3A22730%3Asamba&groupid=284&flavor=Server-DVD-HA-Incidents-Install

### CODE REVIEW

File: lib/utils.pm [hunk 1]
```diff
@@ -716,7 +716,11 @@ sub ssh_fully_patch_system {
     # first run, possible update of packager -- exit code 103
     my $ret = script_run("ssh $remote 'sudo zypper -n patch --with-interactive -l'", 1500);
     record_info('zypper patch', 'The command zypper patch took ' . (time() - $cmd_time) . ' seconds.');
-    die "Zypper failed with $ret" if ($ret != 0 && $ret != 102 && $ret != 103);
+    if (is_sle('=12-SP5') && $ret == 8) {
+        record_soft_failure 'bsc#1195351';
```
Comment (pdostal) [inline lib/utils.pm:720]:
```suggestion
        record_soft_failure 'bsc#1195351 - python3 vs python36 in SLE12 has file conflicts on /usr/bin/pip3';
```

File: tests/qam-updinstall/update_install.pm [hunk 2]
```diff
@@ -145,7 +145,11 @@ sub run {
 
     # Install released version of installable binaries.
     if (scalar(keys %installable)) {
-        zypper_call("in -l " . join(' ', keys %installable), exitcode => [0, 102, 103], log => 'prepare.log', timeout => 1500);
+        my $ret = zypper_call("in -l " . join(' ', keys %installable) . '|tee zypper.log', exitcode => [0, 8, 102, 103], log => 'prepare.log', timeout => 1500);
```
Comment (pdostal) [inline tests/qam-updinstall/update_install.pm:148]: Why is there `|tee zypper.log` and then also `log => 'prepare.log'`? Is that correct?

**Review by grisu48 — APPROVED** [review]
I don't like that we have to do this (it's a valid product bug that should be fixed within the product) but I see the need of at some point doing something.

The PR itself looks good. Thanks @dzedro for taking care! :+1:

File: tests/qam-updinstall/update_install.pm [hunk 3]
```diff
@@ -187,7 +191,7 @@ sub run {
 
     # Patch binaries already installed.
     my $ret = zypper_call("in -l -t patch ${patches}|tee zypper.log", exitcode => [0, 8, 102, 103], timeout => 1500);
-    if ($ret == 8 && script_run('grep -z "python36-pip.*SLES:12-SP5.*conflicts with.*python3-pip" zypper.log') == 0) {
+    if ($ret == 8 && script_run('grep -Ez "python3(6?)-pip.*(SLES:12-SP5|cloud:12).*conflicts with.*python3(6?)-pip" zypper.log') == 0) {
         record_soft_failure 'bsc#1195351';
```
Comment (pdostal) [inline tests/qam-updinstall/update_install.pm:195]:
```suggestion
        record_soft_failure 'bsc#1195351 - python3 vs python36 in SLE12 has file conflicts on /usr/bin/pip3';
```

**Review by pdostal — APPROVED** [review]
Thank you Jozef!

File: tests/qam-updinstall/update_install.pm (hunk 2 above)
Comment (dzedro) [reply tests/qam-updinstall/update_install.pm:148]: I will fix that. It's doing the same.

**Comment by dzedro** [comment]
> I don't like that we have to do this (it's a valid product bug that should be fixed within the product) but I see the need of at some point doing something.
> 
> The PR itself looks good. Thanks @dzedro for taking care! +1

**Pull request merged**